test/integration/model/scope/merge.test.js

```diff
@@ -165,6 +165,9 @@ describe(Support.getTestDialectTeaser('Model'), () => {
         const results = await Promise.all(this.scopePermutations.map(([a, b, c, d]) => this.Foo.scope(a, b, c).findOne(this.scopes[d])));
         const first = results.shift().toJSON();
         for (const result of results) {
+          // flaky test - sometimes it gets to:
+          // - bazs: [ { id: 4, quxes: [ qux7, qux8 ] }, { id: 3, quxes: [ qux5, qux6] ] } ]
+          // + bazs: [ { id: 3, quxes: [ qux5, qux6 ] }, { id: 4, quxes: [ qux7, qux8] ] } ]
```

Comment:
If the order doesn't matter, let's sort it by id then

Reply:
I am not sure - we need to be careful between `undefined` or `default behavior desgin`